src/rest/handlers/RequestHandler.js

```diff
@@ -68,34 +68,37 @@ class RequestHandler {
           resolve();
         }
       };
-      item.request.gen().end((err, res) => {
+      item.request.make().then(res => {
         if (res && res.headers) {
           if (res.headers['x-ratelimit-global']) this.manager.globallyRateLimited = true;
           this.limit = Number(res.headers['x-ratelimit-limit']);
           this.resetTime = Date.now() + Number(res.headers['retry-after']);
```

Comment:
`retry-after` doesn't seem to exist, but `x-ratelimit-reset` does.

The inspection of `res.headers` showed this part to me:

```javascript
{
  // ...
  'x-ratelimit-limit': [ '30' ],
  'x-ratelimit-remaining': [ '29' ],
  'x-ratelimit-reset': [ '1528588913' ],
  // ...
}
```

Effectively, `res.headers.get('retry-after')` was returning `null`, whereas `res.headers.get('x-ratelimit-reset')` returned `1528588913` (which is in seconds) in this test.

Reply:
retry-after is only present when you hit the ratelimit.

Reply:
![image](https://user-images.githubusercontent.com/20760160/41197223-6d9b79bc-6c56-11e8-9a3c-af91d907796e.png)


![image](https://user-images.githubusercontent.com/20760160/41197220-5e3d1d0e-6c56-11e8-8d3a-594fab69cb8f.png)


![image](https://user-images.githubusercontent.com/20760160/41197219-57951ff6-6c56-11e8-8f45-523680b984cc.png)

Just by casually spamming 10 messages, only showing 3 screenshots here, you get the picture.

Reply:
In my tests, I have ran a ping command (send + edit) 10 times, trying to reach the ratelimit, but no luck. Perhaps we should do a fallback, but one question, isn't this handled by the `res.status === 429` check? In terms of correctness, and following the [api docs](https://discordapp.com/developers/docs/topics/rate-limits), the ratelimits end after the remaining time:

> **X-RateLimit-Reset** - Epoch time (seconds since 00:00:00 UTC on January 1, 1970) at which the rate limit resets

But as you said, and the API docs confirm, the `retry-after` header only appears on 429, and inside the check we do for it, we re-queue after said time (plus `restTimeOffset`).

I was only bringing it up because we were using `retry-after` instead of `x-ratelimit-reset`, despite of it not being always available.

Reply:
Yea I was actually reading that wrong at first, I was under the assumption you made the claim that it doesn't exist in general.

I actually don't know why it is handled like that. I am thinking the same here.